### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by zapo**
- Fix tests that were broken after adding a new minor_version param to the encoder
- Ignore *.so files
- Fix broken Time comparison in tests, instead use epoch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ pkg
 ext/bert/c/Makefile
 ext/bert/c/*.bundle
 ext/bert/c/*.o
+ext/bert/c/*.so
```
Comment (LittleCatBear) [inline .gitignore:9]: why are we ignoring *.so files ? (What are they?)

File: test/bert_test.rb [hunk 2]
```diff
@@ -57,6 +56,13 @@ class BertTest < Test::Unit::TestCase
       end
     end
 
+    should "do roundtrips for date time" do
+      now = Time.now
```
Comment (LittleCatBear) [inline test/bert_test.rb:60]: I read somewhere recently (might be rubocop) that time using timezone (like `Time.zone.now`)was a better option, does it apply here?

File: test/encoder_test.rb [hunk 3]
```diff
@@ -83,20 +83,20 @@ class EncoderTest < Test::Unit::TestCase
 
     should 'handle utf8 strings' do
       bert = [131, 109, 0, 0, 0, 5, 195, 169, 116, 195, 169].pack('C*')
-      assert_equal bert, BERT::Encoder.encode("été")
+      assert_equal bert, BERT::Encoder.encode("été", 0)
```
Comment (LittleCatBear) [inline test/encoder_test.rb:86]: Why do we need this extra `0` param? Does it break how we use the gem in trader?

File: test/bert_test.rb (hunk 2 above)
Comment (zapo) [reply test/bert_test.rb:60]: It does only in Rails projects. Ruby doesn't have `Time#zone`

File: .gitignore (hunk 1 above)
Comment (zapo) [reply .gitignore:9]: Dynamic library output from building the C extension (similar to .bundle but on linux systems).

File: test/bert_test.rb (hunk 2 above)
Comment (zapo) [reply test/bert_test.rb:60]: The rationale is that using Time without specifying the zone is using server tz or TZ= env which might be different from the Time.zone= set in rails projects.

File: test/encoder_test.rb (hunk 3 above)
Comment (zapo) [reply test/encoder_test.rb:86]: It was introduced to differentiate float encoding some time ago. It defaults to 0 when using `BERT.encode` directly which we do in trader. This argument is actually required when using `Encoder#encode`.

**Review by mjsommer — CHANGES_REQUESTED** [review]
Antoine, we discussed fixing the Ruby 2.4 Bignum and Fixnem deprecation issues, but that does not seem to be included. Is that still coming? I am pulling the code as follows: `gem 'bert', platform: :ruby, :github => 'adgear/bert', :branch => 'fix/tests'`, and still getting the same Bignum and Fixnum issues.

File: test/bert_test.rb (hunk 2 above)
Comment (LittleCatBear) [reply test/bert_test.rb:60]: right, my bad, thanks :)

File: test/encoder_test.rb (hunk 3 above)
Comment (LittleCatBear) [reply test/encoder_test.rb:86]: thanks!

**Comment by zapo** [comment]
@mjsommer Yes but that won't be part of that PR. I'm just fixing the test suite here so that changes like  the one you're talking about can be checked for regressions.

**Review by LittleCatBear — APPROVED** [review]

**Review by mjsommer — APPROVED** [review]

**Pull request merged**